src/Sentry/Protocol/ISpan.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+
+namespace Sentry.Protocol
+{
+    public interface ISpan
+    {
+        SentryId SpanId { get; }
+
+        SentryId? ParentSpanId { get; }
+
+        SentryId TraceId { get; set; }
+
+        DateTimeOffset StartTimestamp { get; set; }
+
+        DateTimeOffset EndTimestamp { get; set; }
+
+        string? Operation { get; set; }
```

Comment:
While empty operation "works", SDKs would be more useful if they make it obvious that this field must be set. A span without "op" is defaulted during ingestion (in Relay); here's an example:

![image](https://user-images.githubusercontent.com/88819/100862057-f5e18800-3492-11eb-9f7a-594aabf7c3da.png)

What I have done in Go is take an explicit `op` parameter in `StartSpan`.

Reply:
@rhcarvalho is the `Name` of Transaction also required?

Reply:
Yes, here in this example the name is `GET /random.png`

![image](https://user-images.githubusercontent.com/88819/100898032-9f8b3e00-34c0-11eb-8413-fe87de4926d2.png)

![image](https://user-images.githubusercontent.com/88819/100898184-c77aa180-34c0-11eb-9643-11406f4ffccb.png)


For automatic instrumentation provided by the SDK for HTTP servers, the transaction name should be `METHOD /path/pattern` and the transaction `op` should be `"http.server"`

Reply:
Awesome, thanks!

Reply:
Interesting, I always had in mind that the transaction name was the URL and op the HTTP type (get/post,...)

Reply:
Sentry groups transactions by name, so the pattern `METHOD /path/pattern` normally gives you the right split.

We have/had some SDK releases that shipped with only `/path/pattern`, and we're moving away from that because either you have different handlers for different HTTP verbs, or simply a `GET /foo` request will have different performance characteristics than `POST /foo`.

In any case, SDKs should have a mechanism for users to provide a custom transaction name. A common use case is being able to translate things like `GET /users/123/token` into a more generic pattern `GET /users/{}/token`, if the SDK cannot do it by itself.